docs/slicing.md

```diff
@@ -0,0 +1,145 @@
+# Slicing
+
+Functions (which includes strings, arrays, etc.) can be called using slice syntax, The syntax is as follows:
+
+```text
+someSet(optionalLowerBoundExpr;optionalUpperBoundExpr;optionalStepSizeExpr)
+```
+
+As shown above, you can define a slice by defining a lower bound, an upper bound, both or none.
+You can also define a `stepSize` that is used to define the increment of the numbers from the lower
+bound to the upper bound. `stepSize` defaults to `1` when it is not defined.
+
+## Usages
+
+Only certain types of sets can be sliced. These sets are:
+
+1. `Array`
+2. `String`
+3. `Byte Array`
+4. `Dictionary`
```

Comment:
Any binary relation with an `@` attribute should support function call syntax. Raise an issue and link here if this isn't the case.

Reply:
I'm currently limiting it to those data types and any `GenericSet` that only contain binary relation with an `@` attribute. Other than that, the expression will fail, I'll include it in the docs.

Reply:
It currently only supports sets that can be converted to `Array`, `String`, `Byte Array`, or `Dictionary`. Anything other than that, the expression will fail. Should I still raise an issue?